github/create_deployment.yaml

```diff
@@ -0,0 +1,111 @@
+        in "*". To bypass checking entirely, pass an empty string.
+      default: "*"
+    - name: PAYLOAD
+      description: |
+        JSON payload with extra information about the deployment.
+      default: ""
+    - name: ENVIRONMENT
+      description: |
+        Name for the target deployment environment (e.g., production, staging, qa).
+    - name: DESCRIPTION
+      description: |
+        Short description of the deployment.
+      default: ""
+  steps:
+    - name: create-deployment
+      env:
+        - name: GITHUBTOKEN
+          valueFrom:
+            secretKeyRef:
+              name: github
+              key: token
+
+      image: registry.access.redhat.com/ubi8/ubi:latest
+      script: |
+        #!/usr/libexec/platform-python
+        import json
+        import os
+        import http.client
+
+        deployment_url = "/repos/$(params.REPO_FULL_NAME)/deployments"
+
+        if "$(params.REQUIRED_CONTEXTS)" == "*":
+            required_contexts = None
+        elif "$(params.REQUIRED_CONTEXTS)" == "":
+            required_contexts = []
+        else:
+            required_contexts = "$(params.REQUIRED_CONTEXTS)".split(",")
+
+        data = {
+            "ref": "$(params.REF)",
+            "task": "$(params.TASK)",
+            "auto_merge": json.loads("$(params.AUTO_MERGE)"),
+            "required_contexts": required_contexts,
+            "payload": "$(params.PAYLOAD)",
+            "environment": "$(params.ENVIRONMENT)",
+            "description": "$(params.DESCRIPTION)" if "$(params.DESCRIPTION)" else None,
+        }
+        data = { k: v for k, v in data.items() if v is not None }
+        print("Sending this data to GitHub: ")
+        print(data)
+
+        conn = http.client.HTTPSConnection("$(params.GITHUB_HOST_URL)")
+        r = conn.request(
+            "POST",
+            deployment_url,
+            body=json.dumps(data),
+            headers={
+                "User-Agent": "TektonCD, the peaceful cat",
+                "Authorization": "Bearer " + os.environ["GITHUBTOKEN"],
+            })
+        resp = conn.getresponse()
+
+        # we only care about a 201 response to signify a deployment was successfully
+        # created, as a merged branch response returns a 202 status but does not
+        # actually create the deployment.
+        # see https://developer.github.com/v3/repos/deployments/#merged-branch-response
+        if resp.status != 201:
+            print("Error: %d" % (resp.status))
+            print(resp.read())
+            exit(1)
+        else:
+            body = json.loads(resp.read().decode())
+            print("GitHub deployment created for "
+                  "$(params.REPO_FULL_NAME): "
+                  "id={} branch={} sha={} environment={}".format(
+                      body["id"],
+                      body["ref"],
+                      body["sha"],
+                      body["environment"]))
```

Comment:
I'm not sure if this is the best way to be logging some useful data from the response. Any suggestions?

Reply:
I'd be interested to see if we hit the same problem we have in the sentry logging ...

Reply:
I don't think there is other way unfortunately but i think it's good enough as seen in the task logs output...